### PR TITLE
Dllloader cleanup round 2

### PR DESCRIPTION
**Pull request by lrusak**
This is a follow up to #23220 

This fixes some windows warnings by initializing the ordinals with `-1UL`. I've also changed the `const char*` to `std::string_view`, improved the for loops and removed strcmp and the need for pointer addition.

I've added some other cleanup that removes a bunch of unused code.

The diff is best looked at with whitespace changes off -> link

## What is the effect on users?

none

### CODE REVIEW

File: xbmc/cores/DllLoader/Win32DllLoader.cpp [hunk 1]
```diff
@@ -36,75 +37,75 @@ struct Export
...
   // libdvdnav + python need this (due to us using dll_putenv() to put stuff only?)
-  { "getenv",                     -1, (void*)dll_getenv         },
-  { "_environ",                   -1, (void*)&dll__environ      },
-  { "_open_osfhandle",            -1, (void*)dll_open_osfhandle },
+  { "getenv",                     -1UL, (void*)dll_getenv         },
+  { "_environ",                   -1UL, (void*)&dll__environ      },
+  { "_open_osfhandle",            -1UL, (void*)dll_open_osfhandle },
 
-  { NULL,                         -1, NULL                      }
+  { NULL,                         -1UL, NULL                      }
```
Comment (thexai) [inline xbmc/cores/DllLoader/Win32DllLoader.cpp:108]: `std::string_view` cannot be initialized to NULL (runtime crash).

I'm not sure if end array with NULL elements is still need, probably can be removed this end line. This was used in while loop:

`while (exp->name)`

but now is used for loop:

`for (const auto& exp : win32_exports)`

Comment (lrusak) [reply xbmc/cores/DllLoader/Win32DllLoader.cpp:108]: Yep, I missed that. Thanks!

**Comment by lrusak** [comment]
jenkins build this please

**Review by thexai — APPROVED** [review]

**Comment by lrusak** [comment]
squashed the fixups.

**Pull request merged**